common/environment.go

```diff
@@ -64,6 +64,15 @@ func (EnvironmentVariable) UserDir() EnvironmentVariable {
 	}
 }
 
+func (EnvironmentVariable) ApplicationID() EnvironmentVariable {
```

Comment:
I'm somewhere in between "this should be a environment variable" and "this should be additional command line parameters under cmd/root.go".

In my imagination, when the thought of "auto login" came along, my assumption was that we'd perform the login in root.go, that way it'd be automatically implemented for whatever new feature, a la "Write once, use it a million times".

Reply:
I'm sure this has to be set in the env. I'd say set these var, and your script will not have to change or fight with keyctl for login. I did not want to change anything in FE for this. Also, this is more a static variable, unlike 'cap-mbps' or 'log-level' which are tuned for  every new instance of AzCopy.

root.go seems tempting. But at that point I'm sure we'd  not differentiate source/dst  or if one of them have a SAS. Not sure about this though, Auto should be last attempt at Authentication right? After we've explored all of SAS, saved OAuth...

Reply:
I think that this might potentially be confusing for the end-user to have both the Application ID environment variable and the login parameter application ID. At least make them interchangeable on login and I think it'll be OK.

---
In reply to: [484205933](https://github.com/Azure/azure-storage-azcopy/pull/1167#discussion_r484205933) [](ancestors = 484205933)

Reply:
@nakulkar-msft  maybe we can add in the description of these env variables some thing like "This only takes effect when AZCOPY_LOGIN_TYPE is set.